pkg/reporter/reporter.go

```diff
@@ -32,15 +37,50 @@ const (
 
 // Params are the parameters for writing reports.
 type Params struct {
+	HasDiff   bool
+	Details   string
 	Dir       string
-	Operation string
 	IsDestroy bool
-	Output    string
-	HasDiff   bool
+	Message   string
+	Operation string
 }
 
 // Reporter defines the minimum interface for a reporter.
 type Reporter interface {
 	// CreateStatus reports the status of a run.
 	CreateStatus(ctx context.Context, status Status, params *Params) error
+
+	// ClearStatus clears any existing statuses that can be removed.
+	ClearStatus(ctx context.Context) error
+}
+
+type ReporterCommand struct {
+	cli.BaseCommand
+	flags.GitHubFlags
+	FlagReporter string
+}
+
+// ResolveReporterClient resolves and generates the reporter client based on the reporter flag.
+func (r *ReporterCommand) ResolveReporterClient(ctx context.Context) (Reporter, error) {
```

Comment:
Is this a bad pattern? I want to make this available to the commands that need it.

Reply:
Need to re-think this. Need a common way for multiple commands to use a factory pattern to resolve a reporter. But the inputs to each reporter are different, but will come from a common set flags to the command.